controllers/runnerdeployment_controller.go

```diff
@@ -143,6 +144,28 @@ func (r *RunnerDeploymentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
 	}
 
+	if !reflect.DeepEqual(newestSet.Spec.Selector, desiredRS.Spec.Selector) {
```

Comment:
AFAIK, and to be clear, `Selector` is immutable in the standard K8s deployment. But we'd prefer making this mutable for our use-case